tabs don't switch after a vertical gesture

# Bug
A "Quick start" code freezes on the selected tab after a vertical gesture on a TabView. The code there uses regular View elements but the same thing happens when you have scrollable components as a child.


## Environment info
Device: Every Android Device (afaik)


Library version: ^2.6.0

## Steps To Reproduce
Play with tabs. Switch with buttons. Switch with gestures. So far it works. Now make a vertical gesture like scrolling. The selected tab should freeze, buttons do not do anything until you change the tab by swiping.


## Reproducible sample code
https://snack.expo.io/@beqa/tab-view-example-not-working

tried to install from this repo `github:react-native-community/react-native-tab-view#942a72d` gives me the follow error: `npm ERR! Command failed: C:\Program Files\Git\cmd\git.EXE checkout 942a72d
npm ERR! error: pathspec '942a72d' did not match any file(s) known to git.`

--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
     "lint": "eslint --ext .js,.ts,.tsx .",
     "release": "release-it",
     "example": "yarn --cwd example",
-    "bootstrap": "yarn && yarn example",
+    "bootstrap": "yarn example && yarn",
     "prepare": "bob build"
   },
   "publishConfig": {

--- a/src/Pager.tsx
+++ b/src/Pager.tsx
@@ -42,6 +42,7 @@ const {
   Clock,
   Value,
   onChange,
+  and,
   abs,
   add,
   block,
@@ -188,7 +189,7 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
   // Current state of the gesture
   private velocityX = new Value(0);
   private gestureX = new Value(0);
-  private gestureState = new Value(State.END);
+  private gestureState = new Value(State.UNDETERMINED);
   private offsetX = new Value(0);
 
   // Current progress of the page (translateX value)
@@ -415,6 +416,7 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
     divide(abs(this.velocityX), this.velocityX),
     0
   );
+
   private extrapolatedPosition = add(
     this.gestureX,
     multiply(
@@ -514,14 +516,20 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
         // Stop animations while we're dragging
         stopClock(this.clock),
       ],
-
-      cond(eq(this.gestureState, State.END), [
+      [
         set(this.isSwiping, FALSE),
         this.transitionTo(
           cond(
-            greaterThan(
-              abs(this.extrapolatedPosition),
-              divide(this.layoutWidth, 2)
+            and(
+              // We should consider velocity and gesture distance only when a swipe ends
+              // The gestureX value will be non-zero when swipe has happened
+              // We check against a minimum distance instead of 0 because `activeOffsetX` doesn't seem to be respected on Android
+              // For other factors such as state update, the velocity and gesture distance don't matter
+              greaterThan(abs(this.gestureX), SWIPE_DISTANCE_MINIMUM),
+              greaterThan(
+                abs(this.extrapolatedPosition),
+                divide(this.layoutWidth, 2)
+              )
             ),
             // For swipe gesture, to calculate the index, determine direction and add to index
             // When the user swipes towards the left, we transition to the next tab
@@ -546,7 +554,7 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
             this.index
           )
         ),
-      ])
+      ]
     ),
     this.progress,
   ]);

